### ISSUE TITLE
Issue verifying Webhook URL

### ISSUE THREAD

**Issue by PetrosTragoudaras** [VivaPayments/API#2306]
Hello,

We are implementing a Webhook for failed transactions but we are unable to verify the webhook URL in the Viva website.
In the test environment seems to work fine but in the production environment we are unable to veriify the url.

We are using TLS 1.2 as suggested, when we do a GET request typing the url to the browser we get the webhook key printed in JSON format on the browser and we are returning a status code "200 OK" but the Viva website still says:

"Error! URL not verified. Please enter a valid, publicly available, URL that conforms to [these guidelines](https://developer.vivawallet.com/web-api-integration/webhooks) (Error code 502)".

Please let me know what extra info you need for any insights on that.
We need the webhook to be implemented so the whole workflow can go live, thank you in advance.

**Comment by PetrosTragoudaras** [comment]
We are not using any plugins, this is a simple viva implementation.

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

Thank you very much for contacting us. Can you please provide us a screenshot of the error when you try to verify your webhook URL? In addition can you send us the URL you are trying to verify? Thank you in advance.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by PetrosTragoudaras** [comment]
![image](https://user-images.githubusercontent.com/97123022/210248703-49857638-0b9e-4ae0-8d61-39500d55a2c2.png)

**Comment by PetrosTragoudaras** [comment]
Thank you for the fast response, here is the URL as well.
https://www.figame.gr/api.json?action=viva-webhook-key&device=viva&deviceid=test

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras, 

Thank you for the prompt response. Can you please confirm that the [verification key](https://developer.vivawallet.com/webhooks-for-payments/#generate-a-webhook-verification-key) was generated by the credentials of the live account?

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by PetrosTragoudaras** [comment]
Goodmorning! Yes the verification key comes back as JSON from basic auth and it is displayed on the browser

**Comment by PetrosTragoudaras** [comment]
Thank you for the User-Agent!
Yes i can send you the Correlation ID, where would we find that?
Is it the response of the server? this is 403. If it is something else i would like some help locating the appropriate information please.

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras, 

Thank you for the prompt answer. [X-Viva-correlationid](https://developer.vivawallet.com/integration-reference/debugging-errors/#x-viva-correlationid-and-x-viva-eventid) can be found in your browser's inspector, in the tab network. Kindly note that this ID will be generated at the time you try to verify the webhook.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by PetrosTragoudaras** [comment]
Hello again, 

Thank you for the information, here is the correlation ID.
X-Viva-CorrelationId: 23-004-2E8BDD87

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

Thank you very much for the additional information. Our developers are investigating your issue. We will let you know as soon as our investigation is complete. Thank you very much for your patience.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

Thank you very much for the patience. We noticed that the webhook URL does not contain the verification key. The URL returns a blank page. 

![image](https://user-images.githubusercontent.com/112542999/210554607-2eba3a6c-b058-4b37-b6f8-3f6d8e4d2ae7.png)

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

We just wanted to follow up on the reply we sent yesterday. We were simply wondering if you read our response.

In any case this ticket will remain active for at least 24 hours, so that you can come back if you wish.

However, if you have any questions, please don't hesitate to reply back here and I'd be happy to help out!

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

This ticket will close for now. Please don't hesitate to raise a new ticket if you need further assistance.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by PetrosTragoudaras** [comment]
Hello again,
Apologies for the delay, the past 3 days were a Greek bank holiday and we were out of office.
We will review the fact that the key does not return for you and we will find out what might be going on because on our tests the key is properly printed in the browser, i will come back to you with details.
Thank you.

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

Thank you for the feedback. We will awaiting your response. Thank you in advance.

Kind regards,
Viva Wallet E-commerce Support Team

**Comment by PetrosTragoudaras** [comment]
Hello and thank you for commenting on the thread.
Apologise for the misunderstanding, the link we provided was used for testing purposes.
This is the link that we use and it is still not verifying with the same errors as mentioned above.

https://www.figame.gr/index.html?action=viva-webhook-key

This one returns the Key but it does not get verified in the Webhook field as initially mentioned.

Thank you in advance for your response

**Comment by VivaWalletCustomerSupport** [comment]
Hello @PetrosTragoudaras,

After the checks we carried out, we found that the provider of your server has placed a restriction rule so that it does not accept API requests that are made from server to server. It only accepts requests made by the browser to the server. When you try to verify the webhook the above rule does not apply as our server sends a command to your server which is rejected.

We would suggest you to contact your provider and ask them to remove the restriction rule regarding API requests that are made from server to server.

Kind regards,
Viva Wallet E-commerce Support Team
